benchmarks/block_store.py

```diff
@@ -0,0 +1,258 @@
+    )
+
+
+with open("clvm_generator.bin", "wb+") as f:
+    clvm_generator = f.read()
+
+
+async def run_add_block_benchmark():
+
+    verbose: bool = "--verbose" in sys.argv
+    db_wrapper: DBWrapper = await setup_db("block-store-benchmark.db")
+
+    # keep track of benchmark total time
+    all_test_time = 0
+
+    prev_block = bytes([0] * 32)
+
+    try:
+        block_store = await BlockStore.create(db_wrapper)
+
+        block_height = 1
+        timestamp = 1631794488
+        weight = 10
+        iters = 123456
+        sp_index = 0
+        deficit = 0
+        sub_slot_iters = 10
+        required_iters = 100
+        transaction_block_counter = 0
+        prev_transaction_block = bytes([0] * 32)
+        prev_transaction_height = 0
+        total_time = 0.0
+
+        if verbose:
+            print("profiling add_full_block", end="")
+
+        for height in range(block_height, block_height + NUM_ITERS):
+
+            header_hash = rand_hash()
+            is_transaction = transaction_block_counter == 0
+            fees = random.randint(0, 150000)
+            farmer_coin, pool_coin = rewards(height)
+            reward_claims_incorporated = [farmer_coin, pool_coin]
+
+            # TODO: increase fidelity by setting these as well
+            finished_challenge_slot_hashes = None
+            finished_infused_challenge_slot_hashes = None
+            finished_reward_slot_hashes = None
+            sub_epoch_summary_included = None
+
+            record = BlockRecord(
+                header_hash,
+                prev_block,
+                height,
+                weight,
+                iters,
+                sp_index,
+                rand_class_group_element(),
+                None if deficit > 3 else rand_class_group_element(),
+                rand_hash(),  # reward_infusion_new_challenge
+                rand_hash(),  # challenge_block_info_hash
+                sub_slot_iters,
+                rand_hash(),  # pool_puzzle_hash
+                rand_hash(),  # farmer_puzzle_hash
+                required_iters,
+                deficit,
+                deficit == 16,
+                prev_transaction_height,
+                timestamp if is_transaction else None,
+                prev_transaction_block if prev_transaction_block != bytes([0] * 32) else None,
+                None if fees == 0 else fees,
+                reward_claims_incorporated,
+                finished_challenge_slot_hashes,
+                finished_infused_challenge_slot_hashes,
+                finished_reward_slot_hashes,
+                sub_epoch_summary_included,
+            )
+
+            has_pool_pk = random.randint(0, 1)
+
+            proof_of_space = ProofOfSpace(
+                rand_hash(),  # challenge
+                rand_g1() if has_pool_pk else None,
+                rand_hash() if not has_pool_pk else None,
+                rand_g1(),  # plot_public_key
+                32,
+                secrets.token_bytes(8 * 32),
```

Comment:
```suggestion
                random.randbits(8 * 32).to_bytes(32, "big"), # not tested, but you get the idea
```

Reply:
this won't reliably produce the desired number of bytes though.. I'll come up with something else